projects/dsp-ui/src/lib/action/pipes/string-transformation/truncate.pipe.ts

```diff
@@ -25,14 +24,18 @@ import { Pipe, PipeTransform } from '@angular/core';
 })
 export class TruncatePipe implements PipeTransform {
 
-    transform(value: string, args: string[]): string {
+    transform(value: string, limit?: number, trail?: string): string {
```

Comment:
I was actually wondering why this works since the interface looks like this:

```ts
export declare interface PipeTransform {
    transform(value: any, ...args: any[]): any;
}
```

So `args` is actually a rest parameter which you can access as an array. However, I tested your implementation from the template and it works.

Reply:
I have found this:

https://github.com/angular/angular/blob/a62c1c441c67616ce5eacc7e8dcfe169316d4552/packages/core/test/acceptance/pipe_spec.ts#L22-L27

Reply:
I wonder if we should test pipes in templates, see https://github.com/angular/angular/blob/a62c1c441c67616ce5eacc7e8dcfe169316d4552/packages/core/test/acceptance/pipe_spec.ts#L86-L99

Reply:
> I have found this:
> 
> https://github.com/angular/angular/blob/a62c1c441c67616ce5eacc7e8dcfe169316d4552/packages/core/test/acceptance/pipe_spec.ts#L22-L27

I believe this is what I'm already doing, right?

Reply:
> I believe this is what I'm already doing, right?

yes

Do you think it makes sense to test the pipe in a template setup:

```ts
 it('should support arguments in pipes', () => {
    @Component({
      template: `{{person.name | multiArgPipe:'one':person.address.city}}`,
    })
    class App {
      person = {name: 'value', address: {city: 'two'}};
    }

    TestBed.configureTestingModule({declarations: [App, MultiArgPipe]});
    const fixture = TestBed.createComponent(App);
    fixture.detectChanges();

    expect(fixture.nativeElement.textContent).toBe('value one two default');
  });
```

https://github.com/angular/angular/blob/a62c1c441c67616ce5eacc7e8dcfe169316d4552/packages/core/test/acceptance/pipe_spec.ts#L86-L99

Reply:
> > I believe this is what I'm already doing, right?
> 
> yes
> 
> Do you think it makes sense to test the pipe in a template setup:
> 
> ```ts
>  it('should support arguments in pipes', () => {
>     @Component({
>       template: `{{person.name | multiArgPipe:'one':person.address.city}}`,
>     })
>     class App {
>       person = {name: 'value', address: {city: 'two'}};
>     }
> 
>     TestBed.configureTestingModule({declarations: [App, MultiArgPipe]});
>     const fixture = TestBed.createComponent(App);
>     fixture.detectChanges();
> 
>     expect(fixture.nativeElement.textContent).toBe('value one two default');
>   });
> ```
> 
> https://github.com/angular/angular/blob/a62c1c441c67616ce5eacc7e8dcfe169316d4552/packages/core/test/acceptance/pipe_spec.ts#L86-L99

Couldn't hurt to add

Reply:
pipe via template tests added in bbd61c9

Reply:
Wow!